Feat: new creatures

devgabrielsborges opened the pull request:
Summary: This pull request merges the creature branch into the main branch, introducing significant improvements and new features related to creature functionalities in the Jungle project.

Commits:

    Commit [a383420](https://github.com/devgabrielsborges/The-Jungle/commit/a3834205d5c1f8e6ab60610500b5f8fd04553e72):
        Message: feat: rename Criature to Creature and add ICreature interface
        Details:
            Renamed Criature class to Creature.
            Implemented the ICreature interface with methods jump() and attack().
            Refactored the constructor and class implementation to align with the new naming convention.
        Files Changed:
            core/src/main/java/io/github/com/ranie_borges/thejungle/model/Creature.java
            core/src/main/java/io/github/com/ranie_borges/thejungle/model/interfaces/ICreature.java

    Commit [6c5bada](https://github.com/devgabrielsborges/The-Jungle/commit/6c5bada7e9203079ef04f7ba69fc5d0f8baf4fde):
        Message: feat: add Bird, Bunny, Cannibal, Deer, Fish, and Lizard classes with basic functionality
        Details:
            Added new classes for different creatures: Bird, Bunny, Cannibal, Deer, Fish, and Lizard.
            Each class includes basic functionalities and adheres to the ICreature interface.

TheDotOs commented:
Entrar em contato com o diretor de design do projeto, ocorreu um erro de comunicação entre o diretor de design do projeto e os desenvolvedores e designers, onde o lagarto e o peixe não são implementados como criaturas do jogo, assim passível para uma atualização nos assets do jogo.

TheDotOs commented:
Remover o método "jump" das criaturas.

devgabrielsborges commented:
This pull request introduces several new creature classes and an event class, as well as renaming an existing class and adding a new interface. The changes primarily focus on expanding the game's creature system and standardizing the creature-related code.

New creature classes:

* Added `Bird` class extending `Creature` with a constructor and an empty `attack` method.
* Added `Bunny` class extending `Creature` with a constructor and an empty `attack` method.
* Added `Cannibal` class extending `Creature` with a constructor and an empty `attack` method.
* Added `Deer` class extending `Creature` with a constructor and an empty `attack` method.
* Added `Fish` class extending `Creature` with a constructor and an empty `attack` method.
* Added `Lizard` class extending `Creature` with a constructor and an empty `attack` method.

Class renaming and interface addition:

* Renamed `Criature` class to `Creature` and updated its constructor and references. [[1]](diffhunk://#diff-76e1742e5fa7b3868f427c15683fd10ebdabeee87632bd286d48540d0ece60ceR5-R14) [[2]](diffhunk://#diff-76e1742e5fa7b3868f427c15683fd10ebdabeee87632bd286d48540d0ece60ceL22-R24)
* Added `ICreature` interface with an `attack` method.

TheDotOs commented:
Excelente!!!! aplicando o merge pull request!!!
